internal/vault/cache_storage.go

```diff
@@ -183,6 +186,9 @@ func (c *defaultClientCacheStorage) Store(ctx context.Context, client ctrlclient
 		return nil, err
 	}
 
+	// TODO we currently don't pass in owner references
```

Comment:
nit: Something like this?
```suggestion
	// TODO: we currently don't pass in owner references to avoid deleting the cache on restart.
```